### PR TITLE
Replicas: declare suspicious replicas by RSE #5906

### LINKED ISSUE #5906: Allow to declare suspicious replicas by RSE and LFN
Motivation
----------
In #5392 `declare_bad_file_replicas` was modified to allow either a list of PFNs (string) or a list of replicas {'scope': <scope>, 'name': <name>, 'rse_id': <rse_id> or "rse": <rse_name>} as inputs. The `declare_suspicious_file_replicas` API does not have the same feature.

Modification
------------
`declare_suspicious_file_replicas` should be modified to have the same feature.

### PR DESCRIPTION
**Pull request by nsmith-**
Closes #5906

### CODE REVIEW

**Comment by nsmith-** [comment]
This will need modification once #5864 is merged.
FYI @ivmfnal

File: lib/rucio/api/replica.py [hunk 1]
```diff
@@ -75,14 +75,37 @@ def declare_bad_file_replicas(replicas, reason, issuer, vo='def', session=None):
     :param session: The database session in use.
     :returns: Dictionary {rse_name -> [list of replicas failed to declare]}
     """
+    return _declare_bad_or_suspicious_file_replicas(replicas, reason, issuer, vo, BadFilesStatus.BAD, session)
 
+
+@transactional_session
+def declare_suspicious_file_replicas(replicas, reason, issuer, vo='def', session=None):
+    """
+    Declare a list of bad replicas.
```
Comment (cserf) [inline lib/rucio/api/replica.py:84]: Declare a list of suspicious replicas.

**Review by cserf — CHANGES_REQUESTED** [review]
Some tests directly related to this change are failing e.g. :
https://github.com/rucio/rucio/actions/runs/3192824637/jobs/5210982485
```

Declaring suspicious file replica: file://fc1c21894834461e8f50f70b9b8acdb0.cern.ch/tmp/rucio_rse/test_fc1c21894834461e8f50f70b9b8acdb0/mock/55/e5/file_CXRFCSDISP
------------------------------ Captured log call -------------------------------
DEBUG    urllib3.connectionpool:connectionpool.py:1001 Starting new HTTPS connection (1): localhost:443
DEBUG    urllib3.connectionpool:connectionpool.py:456 https://localhost:443 "POST /replicas/suspicious HTTP/1.1" 500 76
_____________________ test_client_add_suspicious_replicas ______________________
[gw1] linux -- Python 3.8.10 /usr/bin/python
test_bad_replica.py:273: in test_client_add_suspicious_replicas
    r = replica_client.declare_suspicious_file_replicas(replicas, 'This is a good reason')
../client/replicaclient.py:99: in declare_suspicious_file_replicas
    raise exc_cls(exc_msg)
E   rucio.common.exception.RucioException: An unknown exception occurred.
E   Details: no error information passed (http status code: 500)
```
Can you please have a look and fix them ?

**Comment by nsmith-** [comment]
This PR is a bit stale, I guess.
I think it would be a nice feature, users could for example say a replica is suspicious if a job fails to read it, and some later daemon could automate checksum verification. So I think the issue should stay open, but if the PR implementation is out of date it may be better to be closed and a new one started.

**Comment by bari12** [comment]
I agree, but it would need to be re-written then. Let's re-open when needed/somebody has time to do it.

**Pull request closed without merging**